### PR TITLE
Javascript 2022 custom metrics

### PR DESCRIPTION
**Pull request by kevinfarrugia**
Required for [JavaScript 2022](https://github.com/HTTPArchive/almanac.httparchive.org/issues/2880)

Added custom metric for `queueMicrotask` and for `requestIdleCallback`

Test: https://webpagetest.httparchive.org/result/220528_7D_Y/1/details/#waterfall_view_step

javascript_new | { ... "requestIdleCallback":1,"queueMicrotask":1 }
-- | --

Control test: https://webpagetest.httparchive.org/result/220528_14_X/1/details/#waterfall_view_step1

javascript_new | { ... "requestIdleCallback":0,"queueMicrotask":0 }
-- | --

### CODE REVIEW

File: dist/javascript.js [hunk 1]
```diff
@@ -91,6 +91,13 @@ return JSON.stringify({
...
-    })()
+    })(),
 
+    requestIdleCallback: (() => {
+        return httparchive_observers["requestIdleCallback"];
+    })(),
+
+    queueMicrotask: (() => {
+        return httparchive_observers["queueMicrotask"];
+    })()
```
Comment (rviscomi) [inline dist/javascript.js:102]:
```suggestion
    requestIdleCallback: httparchive_observers['requestIdleCallback'],

    queueMicrotask: httparchive_observers['queueMicrotask']
```

**Review by rviscomi — COMMENTED** [review]
Small suggestion to simplify the metric values.

Generally, I'm on the fence about the need for chapter-specific metrics when we can query all values from [observers.js](https://github.com/HTTPArchive/custom-metrics/blob/main/dist/observers.js). Do you think it's worth maintaining explicit metrics?

**Comment by kevinfarrugia** [comment]
> Generally, I'm on the fence about the need for chapter-specific metrics when we can query all values from [observers.js](https://github.com/HTTPArchive/custom-metrics/blob/main/dist/observers.js). Do you think it's worth maintaining explicit metrics?

I don't think we should maintain chapter-specific observers and I can remove these two custom-metrics. Awaiting your confirmation.

**Comment by rviscomi** [comment]
SGTM, sorry I know you spent time on this

**Pull request closed without merging**